zebra-consensus/src/transaction.rs

```diff
@@ -403,4 +393,40 @@ where
             )),
         }
     }
+
+    fn verify_transparent_inputs_and_outputs(
+        request: &Request,
+        network: Network,
+        inputs: &[transparent::Input],
+        script_verifier: script::Verifier<ZS>,
+    ) -> Result<impl Stream<Item = Result<(), BoxError>>, TransactionError> {
+        let transaction = request.transaction();
+
+        if transaction.is_coinbase() {
+            check::coinbase_tx_no_prevout_joinsplit_spend(&transaction)?;
+
+            Ok(stream::empty().boxed())
+        } else {
+            // feed all of the inputs to the script and shielded verifiers
+            // the script_verifier also checks transparent sighashes, using its own implementation
+            let cached_ffi_transaction = Arc::new(CachedFfiTransaction::new(transaction));
+            let known_utxos = request.known_utxos();
+            let upgrade = request.upgrade(network);
+
+            let verify_requests =
+                stream::iter((0..inputs.len()).into_iter().map(move |input_index| {
+                    script::Request {
+                        upgrade,
+                        known_utxos: known_utxos.clone(),
+                        cached_ffi_transaction: cached_ffi_transaction.clone(),
+                        input_index,
+                    }
+                }));
+
+            Ok(script_verifier
+                .call_all(verify_requests)
+                .unordered()
+                .boxed())
```

Comment:
This should get you a `futures::stream::FuturesUnordered`, if you change the function to return that specific type:
```suggestion
            let verify_requests =
                (0..inputs.len()).into_iter().map(move |input_index| {
                    let request = script::Request {
                        upgrade,
                        known_utxos: known_utxos.clone(),
                        cached_ffi_transaction: cached_ffi_transaction.clone(),
                        input_index,
                    };
                    script_verifier.clone().oneshot(request)
                }).collect();

            Ok(verify_requests)
```

Reply:
Done, thanks! I just tweaked a few things in the final commit :+1: